### PR TITLE
fix(lint): Fix self closing tags

### LINKED ISSUE #7: does not correctly handle <br /> elements
self closed br tags in translations are always considered invalid HTML even if they are in the original string.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Self closing tags do not cause `onclosetag` to be called,
which caused those tags to be considered still open when
the string terminates.

For self closing tags, manually call onclosetag to ensure
the element is popped off the stack and the tests complete
as expected.

fixes #7

@mozilla/fxa-core - r?

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Note, this can be tested against the current fxa-content-server-l10n repo:

1. clone repo
2. <enter fxa-content-server-l10n repo>
3. npm link ../grunt-l10n-lint
4. npm run test

**Comment by shane-tomlinson** [comment]
beautify decided to re-format things, so easiest to review with ?w=1

File: lib/check-translation.js [hunk 1]
```diff
@@ -246,10 +290,16 @@ CheckTranslation.prototype = {
     } catch (err) {
       this._finish(err);
     }
-  }
+    this._lastTagName = tagName;
```
Comment (chenba) [inline lib/check-translation.js]: I cannot find where this assigned value is used.

Comment (shane-tomlinson) [reply lib/check-translation.js]: Ahha! That's because it's not!

File: lib/check-translation.js [hunk 2]
```diff
@@ -246,10 +290,16 @@ CheckTranslation.prototype = {
...
+  },
+
+  _onCloseTagReplacement(tagName) {
```
Comment (chenba) [inline lib/check-translation.js]: nit: not a fan of this name but I also don't have a suggestion.

**Review by chenba — APPROVED** [review]
r+ with comments

**Pull request merged**